### DISCUSSION TITLE
Add Viber to the "Share Buttons" widget.

### DISCUSSION THREAD

**Discussion by Pantelaz** [elementor/elementor#29332] [Feature Request]
### Prerequisites

- [X] I have searched for similar feature requests in both open and closed discussions and cannot find an existing request.
- [X] I have verified that the feature is still missing from the latest stable versions of Elementor or Elementor Pro.

### What problem is your feature request going to solve? Please describe..

In Elementor Pro "Share Buttons" widget, there are plenty of options to choose, so you can share your pages and posts to many platforms, however, Viber is missing.

### Describe the solution you'd like.

Considering WhatsApp is already being there, why not add a Viber as well. I think many people who have strong Viber communities will find it useful. On top of that, Elementor already supports Viber in some other parts of application - for example:
Button widget > Link > Dynamic Tags > Contact URL > Type > Viber

### Additional context

Upon inspecting the Elementor and Elementor Pro plugins, I was able to modify 2 files in total to add this feature, and it took under than 5 minutes.

In **Elementor** plugin we have file '_elementor/assets/lib/share-link/share-link.js_'.
In this file we have a JavaScript object:
`ShareLink.networkTemplates`
Inside this object, we can add:
`viber: 'viber://forward?text={text}\x20{url}',`

**Note:** This file also have minified version, so I had to add it there as well.

In **Elementor Pro** plugin, we have file '_elementor-pro/modules/share-buttons/module.php_'
In this file we have a PHP associative array `$networks` in `Module` class:

Inside this array we should add:
`'viber' => [
			'title' => 'Viber',
		],`
		
I have tested this code and I can verify that it works on android devices and I think that it should also work on iOS.
Viber application must be installed on the device in order for it to work.

Hope this helps!

### Agreement

- [X] I agree that my request may be closed without action if it doesn't meet all the requirements.

**Comment by mkarim1378** [comment]
Is Viber still alive? XD

**Reply by Pantelaz** [reply]
According to what I found over the internet Viber's active user base is somewhere between 1 and 1.5 billion users. Approximately slightly more than half of WhatsApp's active user base. This makes it highly relevant and alive XD

Also, I know quite a few businesses in Press and Travel that are heavily using Viber for their business.
